prov/efa/test/efa_test_getinfo.c

```diff
@@ -0,0 +1,32 @@
+#include "efa_unit_tests.h"
+
+struct ibv_device **__real_ibv_get_device_list(int *num_devices);
+
+/* This function is a bit tricky. Due to the fact
+ * that it is called during provider init, we can't
+ * use cmocka's mock_type and will_return. Thus this function
```

Comment:
So we cannot mock functions that are called during provider init?

Reply:
We can. We just can't change their behavior per unit test. We can do it per binary though.

Reply:
We can mock them, but we need to hard code in their behavior.  Since they are called before the unit tests execute, the unit tests can't define their behavior.